Artificial Intelligence/README.md

```diff
@@ -0,0 +1,47 @@
+## AI challenge assignments for Wealth42
+
+### Introduction
+The goal of this challenge is to test your familiarity with concepts in  Artificial Intelligence (with Python). To read more about the recruitment process and open opportunities at Wealth42, click [here](http://bit.ly/w42-hiring)
+
+*Note: This is a public submission. Any work on this assignment does not bring Wealth42 or its employees any moat; and by design, is made to assess skills we need. Stories provided are unrelated to Wealth42's core business.* 
+
+### User Story for Assignment
+A school administrator has to manage a Teacher's Schedule by balancing the number of lecture and the number of classes he/she has to teach in a week . For a school to run efficiently, it is necessary that the administrator carves out an optimised schedule for teachers as well as the students with the following constraints:
+1.  There are 5 teachers who have specialization in 5 different subjects.
```

Comment:
Constraints:
1.  There are 5 teachers who have specialization in 5 different subjects.
2. Teacher's have preferences for teaching on certain days over others:
You can assume the preferences by yourself by creating a preference map like this (on a range from 0-10, 10 being the highest, this is absolute for every day and not relative to the other days):
```
{
    "teacher1": {
        "Monday": 5,
        "Tuesday": 10,
        ...
    },...
}
```
3. Students are divided into 3 batches
4. Every batch of students must get at least 5 lectures of each subject, every week
5. There are 5 lectures each day for all batches
6. Teachers are not allowed to have more than 2 lectures in a day for a class.

Optimisation objective: For every lecture taken by any teacher, the 'preference score' is allocated as per the teacher's preference mapping to be taking a lecture in that day. The objective is to optimize the sum of this 'preference score' across the timetables of all teachers.

Using this optimisation objective, find the optimal weekly time table for all 3 batches of the students.